### PR TITLE
Merge `develop` 

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
  - Best-effort merge of `develop` into `04712-use-pbj`

### CODE REVIEW

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/hedera-app-spi/build.gradle.kts [hunk 1]
```diff
@@ -34,11 +34,14 @@ configurations.all {
 }
 
 dependencies {
+  implementation(libs.grpc.stub)
```
Comment (rbair23) [inline hedera-node/hedera-app-spi/build.gradle.kts:37]: Are these dependencies actually needed?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java [hunk 2]
```diff
@@ -101,12 +102,13 @@ public void dispatchHandle(
...
         switch (function) {
             case CONSENSUS_CREATE_TOPIC -> dispatchConsensusCreateTopic(
-                    txn.consensusCreateTopic().orElseThrow(), writableStoreFactory, usageLimits);
-            case CONSENSUS_UPDATE_TOPIC -> dispatchConsensusUpdateTopic(txn.consensusUpdateTopic().orElseThrow());
-            case CONSENSUS_DELETE_TOPIC -> dispatchConsensusDeleteTopic(txn.consensusDeleteTopic().orElseThrow());
-            case CONSENSUS_SUBMIT_MESSAGE -> dispatchConsensusSubmitMessage(txn.consensusSubmitMessage().orElseThrow());
+                    txn.getConsensusCreateTopic().orElseThrow(), topicStore, usageLimits);
+            case CONSENSUS_UPDATE_TOPIC -> dispatchConsensusUpdateTopic(txn.getConsensusUpdateTopic().orElseThrow(), topicStore);
+            case CONSENSUS_DELETE_TOPIC -> dispatchConsensusDeleteTopic(txn.getConsensusDeleteTopic().orElseThrow(), topicStore);
+            case CONSENSUS_SUBMIT_MESSAGE -> dispatchConsensusSubmitMessage(txn, topicStore);
```
Comment (rbair23) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java:111]: It looks like these are reverting PBJ changes (getFoo vs. foo method calls)

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java [hunk 3]
```diff
@@ -31,30 +35,32 @@
...
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /** Implementation of {@link PreHandleWorkflow} */
+@Singleton
 public class PreHandleWorkflowImpl implements PreHandleWorkflow {
```
Comment (rbair23) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java:62]: I think you should exclude these changes from the PR. This is too complicated for me to approve right now. The pbj branch makes a lot of changes in here and I'm afraid we're going to get a bad merge. Instead, I will resolve this by hand in the pbj branch (i.e. diff with develop and get it all squared away). If this gets merged as is, then I think I'm going to lose work and have to revert at least this file.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImplTest.java [hunk 4]
```diff
@@ -81,529 +81,533 @@
...
-//            // And the metrics for counting submitted transactions was not incremented
-//            verify(countSubmitted, never()).increment();
-//        }
-//    }
-//
-//    private static TransactionResponse parseResponse(@NonNull final DataBuffer responseBuffer) throws IOException {
-//        responseBuffer.flip();
-//        return TransactionResponse.PROTOBUF.parse(responseBuffer);
-//    }
+    /** The workflow to be tested. */
```
Comment (rbair23) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImplTest.java:84]: I'm worried about this one too. At least include the commented out code + the new code and I can sort it out.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java [hunk 5]
```diff
@@ -69,1059 +69,428 @@
 
 @ExtendWith(MockitoExtension.class)
 class QueryWorkflowImplTest extends AppTestBase {
-//
```
Comment (rbair23) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java:72]: I'm worried about this one too. At least include the commented out code + the new code and I can sort it out.

**Review by rbair23 — CHANGES_REQUESTED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java:111]: yeah agreed ! This should be reverted ?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/fees/MonoFeeAccumulatorTest.java [hunk 6]
```diff
@@ -47,32 +52,51 @@ class MonoFeeAccumulatorTest {
...
 
     @BeforeEach
     void setUp() {
-        subject = new MonoFeeAccumulator(usageBasedFeeCalculator, usagePricesProvider, () -> stateView);
+        subject = new MonoFeeAccumulator(
+                usageBasedFeeCalculator, getTopicInfoUsage, usagePricesProvider, () -> stateView);
+    }
+
+    @Test
+    void usesMonoAdapterDirectlyForGetTopicInfo() {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/fees/MonoFeeAccumulatorTest.java:73]: Should this be kept commented?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/signature/MonoSignaturePreparerTest.java [hunk 7]
```diff
@@ -38,41 +50,135 @@
...
+        assertEquals(List.of(), result.cryptoSigs());
+    }
+
+    @Test
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    void abortsAfterOtherPartyFailureAndIncludesOnlyPayerSigs() {
+        given(keyToSigFactory.apply(any())).willReturn(keyToSigBytes);
+        given(scopedFactoryProvider.apply(any())).willReturn(scopedFactory);
+        givenHappyTwo(payerResult);
+        given(cryptoSigsCreation.createFrom(List.of(PAYER_KEY), keyToSigBytes, scopedFactory))
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/signature/MonoSignaturePreparerTest.java:132]: same here ?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java [hunk 8]
```diff
@@ -69,1059 +69,428 @@
...
+                currentPlatformStatus,
+                stateAccessor,
+                throttleAccumulator,
+                submissionManager,
+                checker,
+                dispatcher,
+                metrics,
+                feeAccumulator,
+                queryContext);
+    }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java:168]: same here ?

**Review by Neeharika-Sompalli — COMMENTED** [review]
LGTM ! Few questions on some tests, if they should be kept commented.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java]: Fixed.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java]: Checked this out from `04712-use-pbj`.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImplTest.java (hunk 4 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImplTest.java:85]: Reverted this to `04712-use-pbj` contents.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java (hunk 5 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java:72]: Reverted this to `04712-use-pbj` contents.

File: hedera-node/hedera-app-spi/build.gradle.kts (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app-spi/build.gradle.kts]: Not sure, deleted.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thank you !

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/fees/MonoFeeAccumulatorTest.java (hunk 6 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/fees/MonoFeeAccumulatorTest.java:73]: I think relative to the overall "compilability" of the branch, this test crosses the un-commented bar. 😁

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/signature/MonoSignaturePreparerTest.java (hunk 7 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/signature/MonoSignaturePreparerTest.java:132]: As above.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java (hunk 8 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java:168]: Reverted this to `04712-use-pbj` contents.

**Review by mhess-swl — APPROVED** [review]

**Review by rbair23 — APPROVED** [review]

**Review by kimbor — APPROVED** [review]

**Pull request merged**